core/payment-driver/erc20/src/service.rs

```diff
@@ -181,6 +184,21 @@ impl Erc20Service {
                         }
                     };
                 }
+                if let Ok(lock_payment_addr) = env::var(&lock_payment_addr_env) {
+                    match H160::from_str(&lock_payment_addr) {
+                        Ok(parsed) => {
+                            log::info!(
+                                "{network} lock payment contract address set to {lock_payment_addr}"
+                            );
+                            chain.lock_contract = Some(LockContractSettings { address: parsed })
+                        }
+                        Err(e) => {
+                            log::warn!(
+                                "Value {lock_payment_addr} for {lock_payment_addr_env} is not valid H160 address: {e}"
+                            );
+                        }
+                    };
+                }
```

Comment:
We should remove a requirement for hard-coding the lock-contract address at all at some point.